eZ/Bundle/EzPublishCoreBundle/Features/Context/ContentContext.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\Features\Context;
+
+use Behat\Behat\Context\Context;
+use Behat\Behat\Context\SnippetAcceptingContext;
+use eZ\Publish\API\Repository\Repository;
+use eZ\Publish\API\Repository\Values\Content\Content;
+use RuntimeException;
+
+class ContentContext implements Context, SnippetAcceptingContext
+{
+    /** @var \eZ\Publish\API\Repository\Values\Content\Content */
+    private $currentContent;
+
+    /** @var \eZ\Publish\API\Repository\Values\Content\Content */
+    private $currentDraft;
+
+    /** @var \eZ\Publish\API\Repository\Repository */
+    private $repository;
+
+    public function __construct(Repository $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    /**
+     * @Given /^I create an article draft$/
+     */
+    public function iCreateAnArticleDraft()
+    {
+        $this->currentDraft = $this->createDraft(
+            'article',
+            [
+                'title' => 'Preview draft ' . date('c'),
+                'intro' => '<?xml version="1.0" encoding="UTF-8"?><section xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:ezxhtml="http://ez.no/xmlns/ezpublish/docbook/xhtml" xmlns:ezcustom="http://ez.no/xmlns/ezpublish/docbook/custom" version="5.0-variant ezpublish-1.0"><para>This is a paragraph.</para></section>',
+            ]
+        );
+    }
+
+    /**
+     * @Given /^I create a draft of an existing content item$/
+     */
+    public function iCreateADraftOfAnExistingContentItem()
+    {
+        $this->currentContent = $this->createContentItem(
+            'folder',
+            ['name' => 'BDD preview test']
+        );
+
+        $this->currentDraft = $this->createDraftForContent($this->currentContent);
+    }
+
+    /**
+     * Uses a content type identifier + a hash of fields values
+     * to create and publish a content item below the root location.
+     *
+     * @param string $contentTypeIdentifier
+     * @param array $fields Hash of field def identifier => field value
+     *
+     * @return Content the created content item.
+     */
+    public function createContentItem($contentTypeIdentifier, array $fields)
+    {
+        $draft = $this->createDraft($contentTypeIdentifier, $fields);
+
+        $this->currentContent = $this->repository->sudo(
+            function () use ($draft) {
+                return $this->repository->getContentService()->publishVersion($draft->versionInfo);
+            }
+        );
+
+        $this->currentDraft = null;
```

Comment:
'cause when `$this->createDraft()` is called, it stored the created draft as `$this->currentDraft`. We need to reset it, as it is no longer a draft.